### PR TITLE
299 refactor frontend to backend    20

### PR DESCRIPTION
**Pull request by rhit-toftnebj**
Please look over this carefully

### CODE REVIEW

File: apps/frontend/firebase/db.ts
Comment (bveal52) [inline apps/frontend/firebase/db.ts:1]: Need to keep experiment data when converting to json

File: apps/frontend/pages/api/experiments/delete/[expIdToDelete].tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+		return;
+	}
+
+	try {
+		const client = await clientPromise;
+		const db = client.db(DB_NAME);
+
+		const experiment = await db
+            .collection(COLLECTION_EXPERIMENTS)
+            .deleteOne({ '_id': expIdToDelete as any }); // Assuming expId is the unique identifier in the collection
```
Comment (rhit-windsors) [inline apps/frontend/pages/api/experiments/delete/[expIdToDelete].tsx:18]: Make sure this actually returns what you are expecting.

File: apps/frontend/pages/api/experiments/storeExp.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    } catch (error) {
+        const message = 'Failed to store the experiment data';
+        console.error('Error contacting server: ', error);
+        res.status(500).json({ response: message } as any);
+    }
+};
+
+// TODO: In DispatchStep.tsx:
+// fetch('/api/storage/storeExp', {
+//   method: 'POST',
```
Comment (bveal52) [inline apps/frontend/pages/api/experiments/storeExp.tsx:41]: cleanup comments :)

**Review by bveal52 — CHANGES_REQUESTED** [review]
fix the typing, then will approve

File: apps/frontend/pages/api/experiments/storeExp.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import { NextApiHandler } from 'next';
+import { ExperimentData } from '../../../firebase/db_types';
+
+const mongoExpHandler: NextApiHandler = async (req, res) => {
+    if (req.method !== 'POST') {
+        res.status(405).json({ response: 'Method not allowed, use POST' } as any);
+        return;
+    }
+
+    const experimentData: ExperimentData = req.body;
```
Comment (rhit-windsors) [inline apps/frontend/pages/api/experiments/storeExp.tsx:11]: Is this actually ExperimentData? or just a JSON?

**Review by rhit-windsors — CHANGES_REQUESTED** [review]

**Pull request closed without merging**